docs/modules/ROOT/pages/headers.adoc

```diff
@@ -0,0 +1,54 @@
+# DataSonnet Header
+
+The header is an optional comment section in the mapping document which must start with `/\** DataSonnet` followed by a new line character and end with a line `*/` and contains a set of configuration properties.
+The properties are hierarchical, with sections separated by the `.` (dot) character, e.g.:
+
+------------
+dataformat.application/xml.payload.NamespaceSeparator=:
+dataformat.application/xml.payload.TextValueKey=__text
+dataformat.application/csv.myVar.Quote="
+dataformat.application/csv.myVar.Separator=|
+------------
+
+The header must be the first comment before any other line of code, in the following format:
+
+-------------
+/** DataSonnet
+version=<VERSION_NUMBER>
+<PROPERTY>=<VALUE>
+...
+*/
+-------------
+*Example*
+------------------------
+/** DataSonnet
+version=1.0
+*/
+//DataSonnet mapping starts here
+{
+    mappedValue: payload.value
+}
+------------------------
+
+The header is optional; if omitted, all default settings will be used.
+
+## Data Format Properties
+
+The data format properties control the data format plugins behavior. The property names must start with `dataformat.<MIME_TYPE>.payload|output|*|<VAR_NAME>` where `<MIME_TYPE>` is the mime type supported by the DataSonnet. If a property is not recognized, an error will be raised.
+
+IMPORTANT: if the mime type contains the dot (`.`) character, it must be escaped with double backslash (`\\`). For example, `dataformat.application/vnd\\.ms-excel.payload.param=xyz`.
+
+The wildcard `*` is used for default values, if a parameter for payload, output, or variable is not defned explicitly, for example:
```

Comment:
Does this include changing the default values for calls inside the program? I'd be inclined to say yes, but either way we definitely need to document it clearly.

Reply:
Hmm, not sure what you mean by "inside the program". Perhaps I need to rephrase it. The wildcard is used when I want the same behavior for all my inputs and output, e.g.
dataformat.application/csv.*.Separator=| means all CSV inputs are expected to have | as a separator and output will also be separated by |.